test/integration/consul-container/test/gateways/gateway_endpoint_test.go

```diff
@@ -63,32 +76,48 @@ func TestAPIGatewayCreate(t *testing.T) {
 
 	require.NoError(t, cluster.ConfigEntryWrite(apiGateway))
 
+	_, _, err := libservice.CreateAndRegisterStaticServerAndSidecar(cluster.Agents[0], &libservice.ServiceOpts{
+		ID:        serviceName,
+		Name:      serviceName,
+		Namespace: namespace,
+		HTTPPort:  serviceHTTPPort,
+		GRPCPort:  serviceGRPCPort,
```

Comment:
Are these actually needed/being checked in this test? It looks like it was originally simpler, just verifying gateway creation?

Reply:
So, I moved these out to ensure nothing was causing port collisions somewhere. Previously it had pinned the ports to 8080 and 8081 which made running multiple services (i.e. the flatenning code) problematic. With the way that the services get run, the port collision actually was causing the services to get routed improperly because they're competing for the same binding port. I'm assuming the tests were passing in a flaky way before due to sometimes one proxy starting up quicker than another.